src/components/UsernameInput/UsernameInput.tsx

```diff
@@ -13,19 +14,30 @@ interface UsernameInputProps extends Omit<TextStyleProps, "onColorChange"> {
 }
 
 const UsernameInput: FC<UsernameInputProps> = ({ state, onInputChange, label }) => {
-    const handleButtonChange = (options: { isBold?: boolean; isItalics?: boolean }) => {
+    const { setColor } = useSignatureContext();
```

Comment:
We want to keep this component completely generic, with this we would be tightly coupling it to the provider 

We should leverage the same `onInputChange` method that updates the complex state behind the scenes, and it should behave very similar to the `handleInputChange` method 

Something like 

```javascript
onInputChange({ variant: state.variant, color: color })
```

then inside the provider, we do something like 

```javascript
               newOptions.style = {
                    fontWeight: newOptions.isBold ? "bold" : "normal",
                    fontStyle: newOptions.isItalics ? "italic" : "normal",
                    color: newOptions.color ? newOptions.color.toString("hex") : "black"
                };
```

Notice how we do the `toString` at the very end of the process to maintain the integrity of the `Color` interface, if we did it at the `onChange` level, by the time it gets to the provider it would be a string making it hard to manipulate as a color.